modules/geo-layers/src/tile-layer/tileset-2d.js

```diff
@@ -107,6 +104,15 @@ export default class Tileset2D {
         tile.abort();
       }
     }
+    this._cache.clear();
+    this._tiles = [];
+    this._selectedTiles = null;
+  }
+
+  reloadAll() {
+    for (const tile of this._cache.values()) {
+      tile.setNeedsReload();
+    }
```

Comment:
Do you think it makes sense to have two different loops here, one for reloading `_selectedTiles` i.e those in the viewport, and then another for clearing the cache of those outside the viewport?  For example if you zoom out after the load has occurred, you will get old tiles sometimes.  I think it would be better to have just no tiles outside the viewport instead of old data.  I think that was the idea originally with my (very bad) code: https://github.com/visgl/deck.gl/pull/6194/commits/8f4cdd2b7fdaa6d4ac4591c8e0026874d96f5cdd#diff-ebfc47135e3d8fe2bee8b3d33fae4ebbbc295d01887c751d4cb77f71fc85a87aR70-R77

Reply:
```suggestion
  reloadAll() {
    for (const [tileId, tile] of this._cache) {
      if (!this._selectedTiles.includes(tile)) {
        this._cache.delete(tileId);
      } else {
        tile.setNeedsReload();
      }
    }
```
Something like this maybe?

Reply:
Yes the proposed behavior makes sense. Feel free to push a commit.

I'm always a little wary about deleting items in a traversal. `for (const tileId of this._cache.keys())` maybe?

Reply:
updated!